.github/workflows/alpha-deployment.yml

```diff
@@ -0,0 +1,182 @@
+name: Deploy alpha
+
+on:
+  repository_dispatch:
+    types: [ ci-passed ]
+  workflow_dispatch:       # Allow triggering manually
+    inputs:
+      confirmDeployWithoutCI:
+        description: 'I confirm that I want to deploy to ALPHA without running CI on the alpha branch first.'
+        required: true
+        type: boolean
+
+jobs:
+  select-env-to-deploy:
+    name: Prepare metadata for the branch to deploy
+    if: ${{ (inputs.confirmDeployWithoutCI == true) || (github.event.client_payload.ref == 'refs/heads/alpha') }}
+    runs-on: ubuntu-latest
+    outputs:
+      to-deploy: ${{ steps.to-deploy.outputs.list }}
+    steps:
+
+      - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # tag=v3
+        with:
+          ref: alpha
+
+      - name: Construct metadata for the alpha branch deployment
+        id: deployment-candidates
+        # create a list of JSON objects like:
+        # [
+        #  {"name": "alpha", "env": "alpha", "sha": "..."}
+        # ]
+        run: |
+          ALPHA=$(printf '[{"name":"alpha","env":"alpha","sha":"%s"}]' $(git rev-parse alpha))
+          echo "Deployment candidates: $ALPHA"
+          echo ::set-output name=list::$ALPHA
+
+      - name: Read the list of all currently active deployments
+        id: current-deployments
+        run: |
+          # Setup authentication
+          mkdir ~/.kube && echo '${{ secrets.KUBECONFIG }}' > ~/.kube/config && chmod go-r ~/.kube/config
+          # Read list of deployments using helm, and remove the leading ecamp3- from each of the names.
+          # Creates a list of objects like [{"name": "alpha", "sha": "..."}]
+          LIST=$(helm list -o json | jq 'map(.name|=sub("^ecamp3-";""))' | jq 'map({name:.name,sha:.app_version})')
+          echo "Currently active deployments: $LIST"
+          echo ::set-output name=list::$LIST
+
+      - name: Check if the alpha deployment is up to date with the branch status
+        id: to-deploy
+        env:
+          deployments: ${{ steps.current-deployments.outputs.list }}
+          prs: ${{ steps.deployment-candidates.outputs.list }}
+        run: |
+          TO_INSTALL=$(jq --null-input --argjson prs '${{ env.prs }}' --argjson deployments '${{ env.deployments }}' \
+            '$prs|map(select([{name:.name,sha:.sha}]|inside($deployments)|not))')
+          echo "Will install the following candidates, because they either aren't deployed or their deployment is out of date: $TO_INSTALL"
+          echo ::set-output name=list::$TO_INSTALL
+
+  upgrade-or-install-deployment:
+    name: Upgrade or install deployment
+    runs-on: ubuntu-latest
+    needs:
+      - select-env-to-deploy
+    if: fromJSON(needs.select-env-to-deploy.outputs.to-deploy)[0] != null
+    strategy:
+      fail-fast: false
+      matrix:
```

Comment:
do we need a matrix for that?

Reply:
Not really. I've simplified the workflow now.